src/Models/Constants/Interfaces.ts

```diff
@@ -141,10 +142,26 @@ export interface IConsumeCompositeCardProps extends ICardBaseProps {
 }
 
 export interface IAuthService {
-    login: () => void;
-    getToken: (
-        tokenFor?: 'azureManagement' | 'adx' | 'storage' | 'powerBI'
-    ) => Promise<string>;
+    /** the unique object id of the logged in user */
+    userObjectId?: string;
+    setActiveAccount: (account: any) => void;
```

Comment:
I think changing this interface breaks anyone that has implemented their own AuthService. I'm not suggesting we shouldn't change it, but all uses of 3D Scenes I have been involved with do this, so maybe we want to give a heads up.

Reply:
We have various issues with our implementation of IAuthService which was based on the old cardboard code. We have started rewriting it over the next few days but will probably wait for this PR to pick up the new IAuthService definition.

1. When logged into a default domain (eg corpnet), attempting to auth against another directory where this user is a guest with the same email address, we get a ton of auth timeouts and popup dialogs. Eventually it does usually seem to work but its messy.
2. After a while of being logged in, the stream of popups occurs again (presumably refreshing a token or something).
3. Usually after getting this sequence a few times, we get browser error 431 (form fields too long) and the only fix is to restart the browser or clear the cache
4. We don't really want to use popups at all as all our users have to enable popups for the site. Preferable would be one redirect call at login that fetches all the scopes we are going to need at once (maybe pass in the scopes we want  as a login() parameter). The app would then only ever have one token to manage, so the issue of avoiding multiple simultaneous token requests goes away, dramatically simplifying the code. The getToken call would then just refresh "the" single super-token.
5. In our implementation, all the variables used for auth are set in a .env file (eg authority, tenant id, client id etc.). Its not clear to me if this PR supports that or not

Any new auth code should be tested with a parallel AAD directory where the user is a guest, but also where their login is different (select login account).

Reply:
I changed the IAuthService not to break the current implementations with latest commit. You can review it and we can talk about all other breaking changes on Monday. Thanks for listing all the points!